src/battle_util.c

```diff
@@ -9586,7 +9590,7 @@ static u32 CalcFinalDmg(u32 dmg, u16 move, u8 battlerAtk, u8 battlerDef, u8 move
     }
     else if (IsBattlerWeatherAffected(battlerAtk, B_WEATHER_SUN))
     {
-        if (moveType == TYPE_FIRE)
+        if (moveType == TYPE_FIRE || gBattleMoves[move].effect == EFFECT_HYDRO_STEAM)
```

Comment:
The check for Utility Umbrella is still missing. If the mon using Hydro Steam is holding one, then Hydro Steam's damage is halved, not doubled (see [this post](https://www.smogon.com/forums/threads/scarlet-violet-battle-mechanics-research.3709545/page-29#post-9527435)).
Most likely this was overlooked because two lines above the code is incorrect. It should be `(IsBattlerWeatherAffected(battlerDef, B_WEATHER_SUN))`, not `(IsBattlerWeatherAffected(battlerAtk, B_WEATHER_SUN))`.

Reply:
The check for the Utility Umbrella is in `IsBattlerWeatherAffected`. However, when the Utility Umbrella was implemented in the expansion, we all assumed that the defender's Umbrella didn't affect the attacker's damage multiplayer, which Anubis states it does. As discussed in the Discord server, we'll add a new function specific for weather damage multipliers.

Reply:
Addressed in #2835